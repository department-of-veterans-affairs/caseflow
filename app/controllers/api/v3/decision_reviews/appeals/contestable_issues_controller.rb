# frozen_string_literal: true

module Api
  module V3
    module DecisionReviews
      module Appeals
        class ContestableIssuesController < BaseContestableIssuesController
          include ApiV3FeatureToggleConcern

          before_action only: [:index] do
            api_released?(:api_v3_appeals_contestable_issues)
          end

          private

          def standin_decision_review
            @standin_decision_review ||= Appeal.new(
              veteran_file_number: veteran.file_number,
              receipt_date: receipt_date
            )
          end
        end
      end
    end
  end
end
