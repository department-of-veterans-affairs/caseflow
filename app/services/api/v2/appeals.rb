# frozen_string_literal: true

class Api::V2::Appeals
  def initialize(veteran_file_number:, vbms_id:)
    @veteran_file_number = veteran_file_number
    @vbms_id = vbms_id
  end

  def to_hash
    all_reviews_and_appeals
  end

  private

  attr_reader :veteran_file_number, :vbms_id

  def legacy_appeals
    # Appeals API is currently limited to VBA appeals
    @legacy_appeals ||= AppealHistory.for_api(vbms_id: vbms_id).select do |series|
      series.aoj == :vba
    end
  end

  def hlrs
    @hlrs ||= HigherLevelReview.where(veteran_file_number: veteran_file_number).select { |hlr| hlr.request_issues.any? }
  end

  def supplemental_claims
    # Filter out remanded SC because status and information of those are display through
    # the original HLR or Appeal
    @supplemental_claims ||= SupplementalClaim.where(veteran_file_number: veteran_file_number)
      .where(decision_review_remanded: nil)
      .select { |sc| sc.request_issues.any? }
  end

  def appeals
    @appeals ||= Appeal.where(veteran_file_number: veteran_file_number).select { |appeal| appeal.request_issues.any? }
  end

  def all_reviews_and_appeals
    hlr_json = ::V2::HLRStatusSerializer.new(hlrs, is_collection: true).serializable_hash
    sc_json = ::V2::SCStatusSerializer.new(supplemental_claims, is_collection: true).serializable_hash
    appeal_json = ::V2::AppealStatusSerializer.new(appeals, is_collection: true).serializable_hash
    legacy_appeal_json = ::V2::LegacyAppealStatusSerializer.new(legacy_appeals, is_collection: true).serializable_hash

    { data: hlr_json[:data] + sc_json[:data] + appeal_json[:data] + legacy_appeal_json[:data] }
  end
end
