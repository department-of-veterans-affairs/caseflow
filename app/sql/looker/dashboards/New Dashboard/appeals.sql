SELECT * FROM (
SELECT *, DENSE_RANK() OVER (ORDER BY z___min_rank) as z___pivot_row_rank, RANK() OVER (PARTITION BY z__pivot_col_rank ORDER BY z___min_rank) as z__pivot_col_ordering FROM (
SELECT *, MIN(z___rank) OVER (PARTITION BY "decision_documents.bva_decision_dispatched","decision_issues.disposition_date") as z___min_rank FROM (
SELECT *, RANK() OVER (ORDER BY CASE WHEN z__pivot_col_rank=1 THEN 1 ELSE 2 END, CASE WHEN z__pivot_col_rank=1 THEN "appeals.count" ELSE NULL END DESC NULLS LAST, "appeals.count" DESC, z__pivot_col_rank, "decision_documents.bva_decision_dispatched", "decision_issues.disposition_date") AS z___rank FROM (
SELECT *, DENSE_RANK() OVER (ORDER BY "decision_issues.disposition" NULLS LAST, "tasks.status" NULLS LAST) AS z__pivot_col_rank FROM (
SELECT 
	decision_issues.disposition  AS "decision_issues.disposition",
	CASE tasks.status
          WHEN 'assigned' THEN '1_assigned'
          WHEN 'in_progress' THEN '2_viewed'
          WHEN 'on_hold' THEN '3_on_hold'
          WHEN 'completed' THEN '3_completed'
          ELSE tasks.status
         END
         AS "tasks.status",
	CASE WHEN decision_documents.citation_number IS NOT NULL   THEN 'Yes' ELSE 'No' END
 AS "decision_documents.bva_decision_dispatched",
	decision_issues.disposition_date  AS "decision_issues.disposition_date",
	COUNT(DISTINCT appeals.id ) AS "appeals.count"
FROM public.appeals  AS appeals
LEFT JOIN public.tasks  AS tasks ON tasks.appeal_id = appeals.id AND tasks.appeal_type = 'Appeal' 
LEFT JOIN public.decision_issues  AS decision_issues ON appeals.id = decision_issues.decision_review_id AND decision_issues.decision_review_type = 'Appeal' 
LEFT JOIN public.decision_documents  AS decision_documents ON decision_documents.appeal_id = appeals.id 

WHERE ((((decision_documents.decision_date ) >= ((DATE(DATEADD(month,-1, DATE_TRUNC('month', DATE_TRUNC('day',GETDATE())) )))) AND (decision_documents.decision_date ) < ((DATE(DATEADD(month,2, DATEADD(month,-1, DATE_TRUNC('month', DATE_TRUNC('day',GETDATE())) ) ))))))) AND ((((decision_documents.decision_date ) >= ((DATE(DATE_TRUNC('year', DATE_TRUNC('day',GETDATE()))))) AND (decision_documents.decision_date ) < ((DATE(DATEADD(year,1, DATE_TRUNC('year', DATE_TRUNC('day',GETDATE())) )))))))
GROUP BY 1,2,3,4) ww
) bb WHERE z__pivot_col_rank <= 16384
) aa
) xx
) zz
 WHERE z___pivot_row_rank <= 500 OR z__pivot_col_ordering = 1 ORDER BY z___pivot_row_rank