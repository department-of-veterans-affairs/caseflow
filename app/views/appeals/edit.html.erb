<% content_for :full_page_content do %>
  <%= react_component("IntakeEdit", props: {
    userDisplayName: current_user.display_name,
    dropdownUrls: dropdown_urls,
    feedbackUrl: feedback_url,
    buildDate: build_date,
    serverIntake: appeal.ui_hash,
    claimId: url_appeal_uuid,
    featureToggles: {
      useAmaActivationDate: FeatureToggle.enabled?(:use_ama_activation_date, user: current_user),
      # withdrawDecisionReviews: FeatureToggle.enabled?(:withdraw_decision_review, user: current_user),
      editContentionText: FeatureToggle.enabled?(:edit_contention_text, user: current_user),
      correctClaimReviews: FeatureToggle.enabled?(:correct_claim_reviews, user: current_user)
    }
  }) %>
<% end %>
