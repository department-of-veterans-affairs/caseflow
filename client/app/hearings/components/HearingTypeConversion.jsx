import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { get } from 'lodash';
import { sprintf } from 'sprintf-js';
import { withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import React, { useState } from 'react';

import { HearingTypeConversionForm } from './HearingTypeConversionForm';
import { appealWithDetailSelector, taskById } from '../../queue/selectors';
import { deleteAppeal } from '../../queue/QueueActions';
import {
  showErrorMessage,
  showSuccessMessage
} from '../../queue/uiReducer/uiActions';
import ApiUtil from '../../util/ApiUtil';
import COPY from '../../../COPY';
import HEARING_REQUEST_TYPES from
  '../../../constants/HEARING_REQUEST_TYPES';
import TASK_STATUSES from '../../../constants/TASK_STATUSES';

export const HearingTypeConversion = ({
  appeal,
  history,
  task,
  type,
  ...props
}) => {
  // Create and manage the loading state
  const [loading, setLoading] = useState(false);

  const getSuccessMsg = () => {
    const title = sprintf(
      COPY.CONVERT_HEARING_TYPE_SUCCESS,
      appeal?.appellantIsNotVeteran ? appeal?.appellantFullName : appeal?.veteranFullName,
      type
    );
    const detail = sprintf(
      COPY.CONVERT_HEARING_TYPE_SUCCESS_DETAIL,
      appeal?.closestRegionalOfficeLabel || COPY.CONVERT_HEARING_TYPE_DEFAULT_REGIONAL_OFFICE_TEXT
    );

    return { title, detail };
  };

  const submit = async () => {
    try {
      const changedRequestType = type === 'Virtual' ? HEARING_REQUEST_TYPES.virtual : HEARING_REQUEST_TYPES.video;
      const data = {
        task: {
          status: TASK_STATUSES.completed,
          business_payloads: {
            values: {
              changed_request_type: changedRequestType
            }
          }
        }
      };

      setLoading(true);

      await ApiUtil.patch(`/tasks/${task.taskId}`, { data });

      props.showSuccessMessage(getSuccessMsg());
      props.deleteAppeal(task.externalAppealId);
    } catch (err) {
      const error = get(
        err,
        'response.body.errors[0]',
        {
          title: COPY.DEFAULT_UPDATE_ERROR_MESSAGE_TITLE,
          detail: COPY.DEFAULT_UPDATE_ERROR_MESSAGE_DETAIL
        }
      );

      props.showErrorMessage(error);
    } finally {
      setLoading(false);

      history.push(`/queue/appeals/${appeal.externalId}`);
    }
  };

  return (
    <HearingTypeConversionForm
      appeal={appeal}
      history={history}
      isLoading={loading}
      onCancel={() => history.goBack()}
      onSubmit={submit}
      task={task}
      type={type}
    />
  );
};

HearingTypeConversion.propTypes = {
  appeal: PropTypes.object,
  appealId: PropTypes.string,
  deleteAppeal: PropTypes.func,
  showErrorMessage: PropTypes.func,
  showSuccessMessage: PropTypes.func,
  task: PropTypes.object,
  taskId: PropTypes.string,
  type: PropTypes.oneOf(['Virtual']),
  // Router inherited props
  history: PropTypes.object
};

const mapStateToProps = (state, ownProps) => ({
  appeal: appealWithDetailSelector(state, ownProps),
  task: taskById(state, { taskId: ownProps.taskId })
});

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      deleteAppeal,
      showErrorMessage,
      showSuccessMessage
    },
    dispatch
  );

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(HearingTypeConversion)
);
