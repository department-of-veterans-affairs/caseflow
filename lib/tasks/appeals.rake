# frozen_string_literal: true

namespace :appeals do
  class NotEnoughArguments < StandardError; end
  class InvalidLocationPassed < StandardError; end
  class VacolsIdsRequired < StandardError; end

  # usage:
  # Move legacy appeals matching the passed VACOLS ids from/to the passed VACOLS locations (dry run)
  #   $ bundle exec rake appeals:change_vacols_location[55,81,12345,23456,34567,45678]
  # Move legacy appeals matching the passed VACOLS ids from/to the passed VACOLS locations (execute)
  #   $ bundle exec rake appeals:change_vacols_location[55,81,false,12345,23456,34567,45678]
  desc "move legacy appeals from one location to another"
  task :change_vacols_location, [:from_location, :to_location, :dry_run] => :environment do |_, args|
    extras = args.extras
    dry_run = args.dry_run&.to_s&.strip&.upcase != "FALSE"
    if dry_run && args.dry_run.present?
      extras.unshift(args.dry_run)
    end

    logger_tag = "rake appeals:change_vacols_location"
    message = "Invoked with: #{args.to_a.join(', ')}"
    message = "(dry run) " + message if dry_run
    Rails.logger.tagged(logger_tag) { Rails.logger.info(message) }
    puts_output = ""

    # make sure the minimum number of arguments were passed
    if args.to_a.length < 3
      fail NotEnoughArguments,
           "requires at least three arguments: a from location, a to location, and at least one VACOLS ID"
    end

    if extras.empty?
      fail NotEnoughArguments,
           "you must pass VACOLS IDs for the appeals you want to change locations for"
    end

    # make sure valid location codes were passed
    [args.from_location, args.to_location].each do |location|
      if LegacyAppeal::LOCATION_CODES.values.exclude? location
        fail InvalidLocationPassed, "#{location} is not a valid VACOLS location"
      end
    end

    if dry_run
      puts_output += "*** DRY RUN\n"
      puts_output += "*** pass 'false' as the third argument to execute\n"
    end

    # step through the passed vacols IDs
    message = "Changing location for #{extras.length} legacy appeals"
    Rails.logger.tagged(logger_tag) { Rails.logger.info(message) } if !dry_run
    puts_output += message + "\n"

    appeals_moved = 0
    extras.each do |vacols_id|
      la = LegacyAppeal.find_by(vacols_id: vacols_id)
      if la.blank?
        message = "No legacy appeal found for vacols_id #{vacols_id}; skipping."
        Rails.logger.tagged(logger_tag) { Rails.logger.info(message) } if !dry_run
        puts_output += message + "\n"
        next
      end

      # make sure the current location is as expected
      current_location = la.location_code
      if current_location != args.from_location
        message = "Legacy appeal with vacols_id #{vacols_id} is in location " \
                  "#{current_location}, not #{args.from_location}; skipping."
        Rails.logger.tagged(logger_tag) { Rails.logger.info(message) } if !dry_run
        puts_output += message + "\n"
        next
      end

      message = "Moving legacy appeal with vacols_id #{vacols_id} from " \
                "location #{args.from_location} to #{args.to_location}."

      if !dry_run
        AppealRepository.update_location!(la, args.to_location)
        Rails.logger.tagged(logger_tag) { Rails.logger.info(message) }
      end

      puts_output += message + "\n"

      appeals_moved += 1
    end

    message = "Moved #{appeals_moved} of #{extras.length} legacy appeals from " \
              "location #{args.from_location} to #{args.to_location}."
    Rails.logger.tagged(logger_tag) { Rails.logger.info(message) } if !dry_run
    puts_output += message + "\n"
    puts puts_output
  end
end
