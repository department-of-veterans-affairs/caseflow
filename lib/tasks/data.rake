# Requires the following environment variables to be set:
# - VACOLS_PASSWORD
# - VACOLS_USERNAME

namespace :data do
  staging_only_error = "This command only works in the staging environment. Please run with RAILS_ENV='staging'."

  desc "Prepare test data needed to run a smoke test"
  task prepare: [:environment] do
    fail staging_only_error unless Rails.env.staging?

    test_appeal_vacols_id = "2765748"

    test_appeal = Appeal.find_or_create_by_vacols_id(test_appeal_vacols_id)
    AppealRepository.uncertify(test_appeal)
  end
end
