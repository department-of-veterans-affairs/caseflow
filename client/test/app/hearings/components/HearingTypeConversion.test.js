import React from 'react';

import { mount } from 'enzyme';

import {
  HearingTypeConversionForm,
} from '../../../../app/hearings/components/HearingTypeConversionForm';
import { legacyAppealForTravelBoard } from '../../../data/appeals';
import { queueWrapper } from '../../../data/stores/queueStore';
import HearingTypeConversion from
  '../../../../app/hearings/components/HearingTypeConversion';

describe('HearingTypeConversion', () => {
  test('Matches snapshot with default props', () => {
    const hearingTypeConversion = mount(
      <HearingTypeConversion
        appeal={legacyAppealForTravelBoard}
        type={'Virtual'}
      />,
      {
        wrappingComponent: queueWrapper,
      }
    );

    expect(hearingTypeConversion.exists(HearingTypeConversionForm)).toBeTruthy();
    expect(hearingTypeConversion).toMatchSnapshot();
  });
});
