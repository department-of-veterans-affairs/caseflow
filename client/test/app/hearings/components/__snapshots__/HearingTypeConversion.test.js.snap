// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HearingTypeConversion Matches snapshot with default props 1`] = `
<withRouter(Connect(HearingTypeConversion))
  appeal={
    Object {
      "appellantAddress": Object {
        "address_line_1": "9999 MISSION ST",
        "address_line_2": "UBER",
        "address_line_3": "APT 2",
        "city": "SAN FRANCISCO",
        "country": "USA",
        "state": "CA",
        "zip": "94103",
      },
      "appellantFullName": "Abellona Valtas",
      "appellantIsNotVeteran": false,
      "appellantRelationship": "Spouse",
      "assignedAttorney": null,
      "assignedJudge": null,
      "assignedToLocation": "Hearing Admin",
      "attorneyCaseRewriteDetails": Object {
        "note_from_attorney": null,
        "untimely_evidence": null,
      },
      "availableHearingLocations": Array [],
      "canEditDocumentId": false,
      "canEditRequestIssues": false,
      "caseReviewId": null,
      "caseType": "Original",
      "caseflowVeteranId": 541,
      "certificationDate": null,
      "closestRegionalOffice": null,
      "closestRegionalOfficeLabel": "Nashville Regional office",
      "completedHearingOnPreviousAppeal": false,
      "decisionDate": null,
      "decisionIssues": Array [],
      "docketName": "Legacy",
      "docketNumber": "200805-541",
      "documentID": null,
      "externalId": "1234456",
      "hearings": Array [],
      "id": "1",
      "isAdvancedOnDocket": false,
      "isLegacyAppeal": true,
      "isPaperCase": false,
      "issueCount": 0,
      "issues": Array [],
      "nodDate": "2020-08-05",
      "overtime": false,
      "powerOfAttorney": Object {
        "representative_address": Object {
          "address_line_1": "9999 MISSION ST",
          "address_line_2": "UBER",
          "address_line_3": "APT 2",
          "city": "SAN FRANCISCO",
          "country": "USA",
          "state": "CA",
          "zip": "94103",
        },
        "representative_email_address": "tom.brady@caseflow.gov",
        "representative_name": "Attorney McAttorneyFace",
        "representative_type": "Attorney",
      },
      "readableHearingRequestType": "Travel",
      "readableOriginalHearingRequestType": "Travel",
      "regionalOffice": null,
      "removed": false,
      "status": "not_distributed",
      "vacateType": null,
      "veteranFileNumber": "123456789",
      "veteranFullName": "Abellona Valtas",
      "veteranInfo": Object {
        "veteran": Object {
          "address": Object {
            "address_line_1": "1234 Main Street",
            "address_line_2": null,
            "address_line_3": null,
            "city": "Orlando",
            "country": "USA",
            "state": "FL",
            "zip": "12345",
          },
          "date_of_birth": "01/10/1935",
          "date_of_death": null,
          "email_address": "Abellona.Valtas@test.com",
          "full_name": "Abellona Valtas",
          "gender": "M",
        },
      },
      "withdrawn": false,
    }
  }
  type="Virtual"
>
  <Connect(HearingTypeConversion)
    appeal={
      Object {
        "appellantAddress": Object {
          "address_line_1": "9999 MISSION ST",
          "address_line_2": "UBER",
          "address_line_3": "APT 2",
          "city": "SAN FRANCISCO",
          "country": "USA",
          "state": "CA",
          "zip": "94103",
        },
        "appellantFullName": "Abellona Valtas",
        "appellantIsNotVeteran": false,
        "appellantRelationship": "Spouse",
        "assignedAttorney": null,
        "assignedJudge": null,
        "assignedToLocation": "Hearing Admin",
        "attorneyCaseRewriteDetails": Object {
          "note_from_attorney": null,
          "untimely_evidence": null,
        },
        "availableHearingLocations": Array [],
        "canEditDocumentId": false,
        "canEditRequestIssues": false,
        "caseReviewId": null,
        "caseType": "Original",
        "caseflowVeteranId": 541,
        "certificationDate": null,
        "closestRegionalOffice": null,
        "closestRegionalOfficeLabel": "Nashville Regional office",
        "completedHearingOnPreviousAppeal": false,
        "decisionDate": null,
        "decisionIssues": Array [],
        "docketName": "Legacy",
        "docketNumber": "200805-541",
        "documentID": null,
        "externalId": "1234456",
        "hearings": Array [],
        "id": "1",
        "isAdvancedOnDocket": false,
        "isLegacyAppeal": true,
        "isPaperCase": false,
        "issueCount": 0,
        "issues": Array [],
        "nodDate": "2020-08-05",
        "overtime": false,
        "powerOfAttorney": Object {
          "representative_address": Object {
            "address_line_1": "9999 MISSION ST",
            "address_line_2": "UBER",
            "address_line_3": "APT 2",
            "city": "SAN FRANCISCO",
            "country": "USA",
            "state": "CA",
            "zip": "94103",
          },
          "representative_email_address": "tom.brady@caseflow.gov",
          "representative_name": "Attorney McAttorneyFace",
          "representative_type": "Attorney",
        },
        "readableHearingRequestType": "Travel",
        "readableOriginalHearingRequestType": "Travel",
        "regionalOffice": null,
        "removed": false,
        "status": "not_distributed",
        "vacateType": null,
        "veteranFileNumber": "123456789",
        "veteranFullName": "Abellona Valtas",
        "veteranInfo": Object {
          "veteran": Object {
            "address": Object {
              "address_line_1": "1234 Main Street",
              "address_line_2": null,
              "address_line_3": null,
              "city": "Orlando",
              "country": "USA",
              "state": "FL",
              "zip": "12345",
            },
            "date_of_birth": "01/10/1935",
            "date_of_death": null,
            "email_address": "Abellona.Valtas@test.com",
            "full_name": "Abellona Valtas",
            "gender": "M",
          },
        },
        "withdrawn": false,
      }
    }
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
    location={
      Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      }
    }
    match={
      Object {
        "isExact": true,
        "params": Object {},
        "path": "/",
        "url": "/",
      }
    }
    type="Virtual"
  >
    <HearingTypeConversion
      deleteAppeal={[Function]}
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
      location={
        Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        }
      }
      match={
        Object {
          "isExact": true,
          "params": Object {},
          "path": "/",
          "url": "/",
        }
      }
      showErrorMessage={[Function]}
      showSuccessMessage={[Function]}
      type="Virtual"
    >
      <HearingTypeConversionForm
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "canGo": [Function],
            "createHref": [Function],
            "entries": Array [
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
            ],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "index": 0,
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
        isLoading={false}
        onCancel={[Function]}
        onSubmit={[Function]}
        type="Virtual"
      >
        <AppSegment
          filledBackground={true}
        >
          <section
            className="cf-app-segment cf-app-segment--alt"
            role="main"
          >
            <h1
              className="cf-margin-bottom-0"
            >
              Convert Hearing To Virtual
            </h1>
            <p
              dangerouslySetInnerHTML={
                Object {
                  "__html": "The hearing request will appear in the scheduling queue for the appropriate regional office.",
                }
              }
            />
            <HelperText
              label="Review and update the following information in VBMS if needed. Caseflow won't send email notifications until you schedule the hearing."
            >
              <span
                data-css-tgxxqt=""
              >
                Review and update the following information in VBMS if needed. Caseflow won't send email notifications until you schedule the hearing.
              </span>
            </HelperText>
            <AppellantSection
              appellantTitle="Veteran"
              hearing={
                Object {
                  "appellantFullName": undefined,
                  "appellantIsNotVeteran": undefined,
                  "representative": undefined,
                  "representativeType": undefined,
                  "veteranFullName": undefined,
                }
              }
              readOnly={true}
              showDivider={false}
              showMissingEmailAlert={true}
              showOnlyAppellantName={true}
              type="Virtual"
              virtualHearing={
                Object {
                  "appellantEmail": undefined,
                  "representativeEmail": undefined,
                }
              }
            >
              <VirtualHearingSection
                hide={false}
                label="Veteran"
                showDivider={false}
              >
                <div
                  data-css-1jfbwxw=""
                />
                <h2>
                  Veteran
                </h2>
                <ReadOnly
                  label="Veteran Name"
                  text="undefined undefined"
                >
                  <div
                    data-css-1a5irm0=""
                  >
                    <strong>
                      Veteran Name
                    </strong>
                    <pre>
                      undefined undefined
                    </pre>
                  </div>
                </ReadOnly>
                <div
                  className="usa-grid css-1jfbwxw"
                  id="email-section"
                >
                  <div
                    className="usa-width-one-half css-uplrl8"
                  >
                    <VirtualHearingEmail
                      emailType="appellantEmail"
                      label="Veteran Email"
                      readOnly={true}
                      required={true}
                      type="Virtual"
                    >
                      <ReadOnly
                        label="Veteran Email"
                        text="None"
                      >
                        <div
                          data-css-1a5irm0=""
                        >
                          <strong>
                            Veteran Email
                          </strong>
                          <pre>
                            None
                          </pre>
                        </div>
                      </ReadOnly>
                    </VirtualHearingEmail>
                    <div>
                      <Alert
                        fixed={false}
                        message="If you have the Veteran's email address, enter it in VBMS. The email will be required to send notifications when you schedule the hearing."
                        scrollOnAlert={false}
                        type="info"
                      >
                        <div
                          className="usa-alert usa-alert-info usa-alert-slim"
                          role="alert"
                        >
                          <div
                            className="usa-alert-body"
                          >
                            <h2
                              className="usa-alert-heading"
                            />
                            <div
                              className="usa-alert-text"
                            >
                              If you have the Veteran's email address, enter it in VBMS. The email will be required to send notifications when you schedule the hearing.
                            </div>
                          </div>
                        </div>
                      </Alert>
                    </div>
                  </div>
                </div>
              </VirtualHearingSection>
            </AppellantSection>
            <RepresentativeSection
              appellantTitle="Veteran"
              hearing={
                Object {
                  "appellantFullName": undefined,
                  "appellantIsNotVeteran": undefined,
                  "representative": undefined,
                  "representativeType": undefined,
                  "veteranFullName": undefined,
                }
              }
              readOnly={true}
              showDivider={false}
              showMissingEmailAlert={true}
              showOnlyAppellantName={true}
              type="Virtual"
              virtualHearing={
                Object {
                  "appellantEmail": undefined,
                  "representativeEmail": undefined,
                }
              }
            >
              <VirtualHearingSection
                hide={false}
                label="Power of Attorney"
                showDivider={true}
              >
                <div
                  className="cf-help-divider"
                />
                <h2>
                  Power of Attorney
                </h2>
                <ReadOnly
                  text="The Veteran does not have a representative recorded in VBMS"
                >
                  <div
                    data-css-1a5irm0=""
                  >
                    <pre>
                      The Veteran does not have a representative recorded in VBMS
                    </pre>
                  </div>
                </ReadOnly>
                <div
                  className="usa-grid css-1jfbwxw"
                >
                  <div
                    className="usa-width-one-half css-uplrl8"
                  >
                    <VirtualHearingEmail
                      emailType="representativeEmail"
                      label="POA/Representative Email"
                      readOnly={true}
                      type="Virtual"
                    >
                      <ReadOnly
                        label="POA/Representative Email"
                        text="None"
                      >
                        <div
                          data-css-1a5irm0=""
                        >
                          <strong>
                            POA/Representative Email
                          </strong>
                          <pre>
                            None
                          </pre>
                        </div>
                      </ReadOnly>
                    </VirtualHearingEmail>
                  </div>
                </div>
              </VirtualHearingSection>
            </RepresentativeSection>
          </section>
        </AppSegment>
        <div
          data-css-1jfbwxw=""
        >
          <Button
            classNames={
              Array [
                "cf-submit",
              ]
            }
            linkStyling={true}
            loading={false}
            name="Cancel"
            onClick={[Function]}
            styling={
              Object {
                "data-css-14x0t8z": "",
              }
            }
            type="button"
            willNeverBeLoading={false}
          >
            <span>
              <button
                className="cf-submit cf-btn-link usa-button"
                data-css-14x0t8z=""
                id="button-Cancel"
                onClick={[Function]}
                type="button"
              >
                Cancel
              </button>
            </span>
          </Button>
          <span
            data-css-jf1dde=""
          >
            <Button
              className="usa-button"
              classNames={
                Array [
                  "cf-submit",
                ]
              }
              linkStyling={false}
              loading={false}
              name="Convert Hearing To Virtual"
              onClick={[Function]}
              type="button"
              willNeverBeLoading={false}
            >
              <span>
                <button
                  className="cf-submit usa-button"
                  id="button-Convert-Hearing-To-Virtual"
                  onClick={[Function]}
                  type="button"
                >
                  Convert Hearing To Virtual
                </button>
              </span>
            </Button>
          </span>
        </div>
      </HearingTypeConversionForm>
    </HearingTypeConversion>
  </Connect(HearingTypeConversion)>
</withRouter(Connect(HearingTypeConversion))>
`;
